infra/templates/dns-records/template.yaml

```diff
@@ -100,10 +104,122 @@ Resources:
         Product: ID Check Async
         System: ID Check Async
         Environment: !Sub ${Environment}
+# New resources below to be kept
+  AsyncExternalCertificate:
+    Type: AWS::CertificateManager::Certificate
+    Properties:
+      DomainName: !Sub
+        - ${DNSSUFFIX}
+        - DNSSUFFIX: !FindInMap
+            - PlatformConfiguration
+            - !Ref Environment
+            - DNSSUFFIX
+      SubjectAlternativeNames:        
+        - !Sub 
+          - sessions-mob-async-backend.${DNSSUFFIX}
+          - DNSSUFFIX: !FindInMap
+              - PlatformConfiguration
+              - !Ref Environment
+              - DNSSUFFIX            
+        - !If
+          - IsBuildOrStagingEnvironment
+          - !Sub
+            - mob-sts-mock.${DNSSUFFIX}
+            - DNSSUFFIX: !FindInMap
+                - PlatformConfiguration
+                - !Ref Environment
+                - DNSSUFFIX
+          - !Ref AWS::NoValue
+        - !If
+          - IsBuildOrStagingEnvironment
+          - !Sub
+            - proxy-mob-async-backend.${DNSSUFFIX}
+            - DNSSUFFIX: !FindInMap
+                - PlatformConfiguration
+                - !Ref Environment
+                - DNSSUFFIX
+          - !Ref AWS::NoValue
+        - !If
+          - IsDevEnvironment
+          - !Sub
+            - '*.${DNSSUFFIX}'
+            - DNSSUFFIX: !FindInMap
+                - PlatformConfiguration
+                - !Ref Environment
+                - DNSSUFFIX
+          - !Ref AWS::NoValue
+      DomainValidationOptions:
```

Comment:
do we need to check the IsDevEnvironment wildcard domain here? i.e. this block from the SubjectAlternativeNames section above:
` - !If
          - IsDevEnvironment
          - !Sub
            - '*.${DNSSUFFIX}'
            - DNSSUFFIX: !FindInMap
                - PlatformConfiguration
                - !Ref Environment
                - DNSSUFFIX
          - !Ref AWS::NoValue`

Reply:
No, I don't think so because it won't be able to validate the domain name as `*.review-b-async.account.gov.uk` does not exist.